community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_2/exceptionHandler.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility.v3_2
+
+import org.neo4j.cypher._
+import org.neo4j.cypher.internal.compatibility.{ExceptionHandler, _}
+import org.neo4j.cypher.internal.frontend.v3_2.spi.MapToPublicExceptions
+import org.neo4j.cypher.internal.frontend.v3_2.{CypherException => InternalCypherException}
+
+object exceptionHandler extends MapToPublicExceptions[CypherException] {
+  def syntaxException(message: String, query: String, offset: Option[Int], cause: Throwable) = new SyntaxException(message, query, offset, cause)
+
+  def arithmeticException(message: String, cause: Throwable) = new ArithmeticException(message, cause)
+
+  def profilerStatisticsNotReadyException(cause: Throwable) = {
+    throw new ProfilerStatisticsNotReadyException(cause)
+  }
+
+  def incomparableValuesException(lhs: String, rhs: String, cause: Throwable) = new IncomparableValuesException(lhs, rhs, cause)
+
+
+  def patternException(message: String, cause: Throwable) = new PatternException(message, cause)
+
+  def invalidArgumentException(message: String, cause: Throwable) = new InvalidArgumentException(message, cause)
+
+  def mergeConstraintConflictException(message: String, cause: Throwable) = new MergeConstraintConflictException(message, cause)
+
+  def internalException(message: String, cause: Exception) = new InternalException(message, cause)
+
+  def loadCsvStatusWrapCypherException(extraInfo: String, cause: InternalCypherException) =
+    new LoadCsvStatusWrapCypherException(extraInfo, cause.mapToPublic(exceptionHandler))
+
+  def loadExternalResourceException(message: String, cause: Throwable) = throw new LoadExternalResourceException(message, cause)
```

Comment:
That is probably a bug. I'll look at older branches and see if the same thing is happening. Good catch!